modules/navier_stokes/src/fvkernels/PINSFVEnergyDiffusion.C

```diff
@@ -66,14 +67,30 @@ PINSFVEnergyDiffusion::computeQpResidual()
     const auto face_elem = elemArg();
     const auto face_neighbor = neighborArg();
 
-    const auto value1 = _porosity_factored_in
-                            ? _k(face_elem, state)
-                            : _k(face_neighbor, state) * _eps(face_neighbor, state);
-    const auto value2 = _porosity_factored_in
-                            ? _k(face_neighbor, state)
-                            : _k(face_neighbor, state) * _eps(face_neighbor, state);
+    auto value1 = _porosity_factored_in ? _k(face_elem, state)
+                                        : _k(face_neighbor, state) * _eps(face_neighbor, state);
+    auto value2 = _porosity_factored_in ? _k(face_neighbor, state)
+                                        : _k(face_neighbor, state) * _eps(face_neighbor, state);
 
-    Moose::FV::interpolate(_k_interp_method, k_eps_face, value1, value2, *_face_info, true);
+    // Adapt to users either passing 0 thermal diffusivity, 0 porosity, or k correlations going
+    // negative. The solution is invalid only for the latter case.
+    auto k_interp_method = _k_interp_method;
+    if (value1 <= 0)
+    {
+      flagInvalidSolution("Negative or null thermal diffusivity value");
+      if (_k_interp_method == Moose::FV::InterpMethod::HarmonicAverage)
+        k_interp_method = Moose::FV::InterpMethod::Average;
+      value1 = 0;
```

Comment:
In debug can we throw a warning here too in case we default back to average? 
Also, could you add a comment on this behavior to the doc? This will affect the solver behavior and the way the solver gets to the final solution.
Why did you change the heat conduction only? I imagine this can be a problem everywhere.

Reply:
Now that I could think about this a bit more: how about we add the following text to the invalidity message:
```
flagInvalidSolution("Negative or null thermal diffusivity value, if this is on purpose use arithmetic mean interpolation instead of the default harmonic interpolation.");
```
The user then can read the documentation and switch the parameter. I am not sure if I am a fan of automatically switching interpolation methods under the hood.

Reply:
> Why did you change the heat conduction only? I imagine this can be a problem everywhere.

Well I havent seen it anywhere else. I m seeing it in a VTB input now though
I think switching interpolation method is a better option than harmonic of two numbers with opposite signs.
Solution invalid is a good option

Reply:
should be addressed

Reply:
Nice! I think we are doing something similar for every diffusion term for the diffusivity. So this can be in every transport equation potentially. Anyways, we can probably address them if we see negative viscosity/diffusivity.